zombies.js

```diff
@@ -78,136 +90,252 @@
- * Player checks their equipment.
- *
- * Prints the player's name and equipped weapon's name.
- * If nothing is equipped, prints a message saying so.
- * Also returns the equipped weapon's name or false if nothing is equipped.
- * You should be able to invoke this function on a Player instance.
- *
- * @name equippedWith
- * @return {string/boolean}   Weapon name or false if nothing is equipped.
- */
-
+class Player {
+  constructor(name, health, strength, speed) {
+    this._name = name;
+    this._health = health;
+    this._strength = strength;
+    this._speed = speed;
+    this._pack = [];
+    this._maxHealth = health;
+    this._isAlive = true;
+    this._equipped = false;
+  }
+
+  get name() {
+    return this._name;
+  }
+
+  get health() {
+    return this._health;
+  }
+
+  set health(health) {
+    if (typeof health === `number`) {
+      this._health = health;
+    }
+  }
+
+  get strength() {
+    return this._strength;
+  }
+
+  get speed() {
+    return this._speed;
+  }
+
+  get equipped() {
+    return this._equipped;
+  }
+
+  get isAlive() {
+    return this._isAlive;
+  }
+
+  getPack() {
+    return this._pack;
+  }
+
+  getMaxHealth() {
+    return this._maxHealth;
+  }
+
+  isItemInPack(item) {
+    if (this.getPack().indexOf(item) === -1) {
+      return false;
+    } else {
+      return this.getPack().indexOf(item);
+    }
+  }
+
+  /**
+   * Player Class Method => checkPack()
+   * -----------------------------
+   * Player checks the contents of their pack.
+   *
+   * Nicely format and print the items in the player's pack.
+   * To access the pack, be sure to use Player's getPack method.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name checkPack
+   */
+  checkPack() {
+    console.log(`Pack:`);
+    let currentPack = this.getPack();
+    currentPack.forEach((curr) => {
+      console.log(`  Item: ${curr.name}`);
+    });
+  }
+
+  /**
+   * Player Class Method => takeItem(item)
+   * -----------------------------
+   * Player takes an item from the world and places it into their pack.
+   *
+   * Player's pack can only hold a maximum of 3 items, so if they try to add more
+   *   than that to the pack, return false.
+   * Before returning true or false, print a message containing the player's
+   *   name and item's name if successful.  Otherwise, print a message saying
+   *   that the pack is full so the item could not be stored.
+   * Note: The player is allowed to store similar items (items with the same name).
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name takeItem
+   * @param {Item/Weapon/Food} item   The item to take.
+   * @return {boolean} true/false     Whether player was able to store item in pack.
+   */
+  takeItem(item) {
+    if (this.getPack().length >= 3) {
+      console.log(`Pack is full!`);
+      return false;
+    } else {
+      console.log(`Player: ${this._name}, Item: ${item.name}`);
```

Comment:
You actually setup an accessor for name, so you could use `this.name` and get to `this._name` through it.